python/covid_graphs/covid_graphs/predictions.py

```diff
@@ -241,36 +243,50 @@ def load_prediction_db(prediction_dir: Path) -> PredictionDb:
         ]
         # TODO(miskosz): The decision on which predictions to display should not be a reponsibility
         # of `predictions` module. All data should be served.
-        start_idx = _calculate_earliest_displayable_idx(fitted_formulas, MAX_PEAK_DISTANCE)
+        start_idx, end_idx = _calculate_displayable_interval(
+            fitted_formulas, MAX_PEAK_DISTANCE, MAX_PEAK_VARIABILITY
+        )
         fitted_predictions = _create_predictions_from_formulas(
-            fitted_formulas[start_idx:], country_atg_parameters.short_country_name
+            fitted_formulas[start_idx:end_idx], country_atg_parameters.short_country_name
         )
         country_predictions.extend(fitted_predictions)
 
     return PredictionDb(country_predictions=country_predictions)
 
 
-def _calculate_earliest_displayable_idx(
-    fitted_formulas: Iterable[FittedFormula], max_peak_distance: datetime.timedelta
-) -> int:
+def _calculate_displayable_interval(
+    fitted_formulas: Iterable[FittedFormula],
+    max_peak_distance: datetime.timedelta,
+    max_peak_variability: datetime.timedelta,
+) -> Tuple[int, int]:
     """
-    Calculates the earliest index in fitted_formulas, after which the peaks of formulas are within
-    'max_peak_distance'.
+    Returns a half-open interval [start, end), for which the following holds:
+    * fitted_formulas[end - 1] is the latest prediction whose peak is not too far in the future,
+      i.e. it's earlier than 'max_peak_distance' from now.
+    * fitted_formula[start] is the earliest prediction, after which the peaks are within
+      'max_peak_variability'.
     """
     peak_days = [
         formula.get_peak(country_report=None) for formula in reversed(list(fitted_formulas))
```

Comment:
I would prefer if we did not reverse the array, it makes the code more difficult to follow (you can iterate in reverse order by using `array[from:to:step]` notation with a negative step)

Reply:
I'm thinking that the API might not be the best right now, it was made for the previous use case which lived outside of `prediction.py`.

I suggest
- accepting `FittedFormula`'s in any order and we will sort them internally however we want.
- instead of interval, we will return a subset of `fitted_formulas` with the desired properties.

This will simplify things and we would just construct the list instead of calculating the interval.

Reply:
That will definitely help!

Reply:
I've now changed the API, PTAL.

I'm not inclined to simplify this further, since:
* this is really a TopCoder division 2 500 point problem, maybe not even that
* there are other parts of the code that need love right now :smile: